app.py

```diff
@@ -0,0 +1,136 @@
+from datetime import datetime
+from flask import Flask, render_template, request, request, jsonify, session
+from dotenv import load_dotenv
+from flask_session import Session
+from Infrastructure.domain.entities import models
+import os
+import requests
+import json
+
+load_dotenv()
+
+app = Flask(__name__)
+app.config["SESSION_PERMANENT"] = False
+app.config["SESSION_TYPE"] = "filesystem"
+Session(app)
+
+AUTHORIZE_URL = os.getenv("RSO_BASE_URI") + "/authorize"
+TOKEN_URL = os.getenv("RSO_BASE_URI") + "/token"
+LOGOUT_URL = os.getenv("RSO_BASE_URI") + "/logout"
+APP_CALLBACK_URL = os.getenv("APP_BASE_URL") +":"+ os.getenv("PORT") + os.getenv("APP_CALLBACK_PATH")
+RSO_CLIENT_ID = os.getenv("RSO_CLIENT_ID")
+AUTH_URL=AUTHORIZE_URL + "?redirect_uri=" + APP_CALLBACK_URL + "&client_id=" + RSO_CLIENT_ID + "&response_type=code" + "&scope=openid"
+
+@app.route("/", methods=["POST","GET"])
+def index(): 
+    return render_template("index.html", AUTH_URL=AUTH_URL, LOGOUT_URL=LOGOUT_URL)
+
+@app.route("/oauth-callback", methods=["POST","GET"])
+def aouthCallback():
+    code = request.args.get("code")
+
+    data = {
+    "grant_type": "authorization_code",
+    "code": code,
+    "redirect_uri": APP_CALLBACK_URL
+    }
+
+    try:
+        req = requests.Session()
+        req.auth = (os.getenv("RSO_CLIENT_ID"), os.getenv("RSO_CLIENT_SECRET"))
+
+        r = req.post(
+            url = TOKEN_URL, 
+            data = data, 
+            headers={"Content-Type":"application/x-www-form-urlencoded"},
+            )
+
+        session["token"] = r.json().get("access_token")
+
+        return render_template("main.html", account={})   
+    except Exception as e:
+        print(f"Error token not generated {e}")
```

Comment:
The other exceptions we're just printing raw, did we want to include additional info like this for all of them? If so, maybe:
```
print(f"Token not generated: {e}")
```